### ISSUE TITLE
Potential resource leak?

### ISSUE THREAD

**Issue by charleswhchan** [KushalP/serfclient-py#29]
We have a function that we call to get a list of serf members that are alive:

```
def get_members(tags=None):
        client = SerfClient()
        raw_members = client.members(status='alive', tags=tags)
        ...
```

During my review, I notice SerfClient() doesn't have a close() function call. Since SerfClient [holds a reference](https://github.com/KushalP/serfclient-py/blob/master/serfclient/client.py#L12) to SerfConnection, and that in turn [holds a reference](https://github.com/KushalP/serfclient-py/blob/master/serfclient/connection.py#L37) to socket -- won't this result in a resource leak if we don't cleanup the client/connection/socket properly?

Basically I am looking for a way to release the SerfClient() resource properly. Am I missing anything?

**Comment by charleswhchan** [comment]
Should we call shutdown() and close() on socket?

**Comment by charleswhchan** [comment]
@KushalP: I made an attempt to fix the issue. Can you please review the PR? :smiley:
